### PR TITLE
Add workflow pending list and process approval APIs

### PR DESCRIPTION
**Pull request by Krishanx92**
Fix https://github.com/wso2/apk/issues/1168
Fix https://github.com/wso2/apk/issues/1169

### CODE REVIEW

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+import wso2/apk_common_lib as commons;
+import ballerinax/postgresql;
+import ballerina/sql;
+
+//This function is used to retrive the pending workflow requests 
+// Using Workflow table
+isolated function getWorkflowListDAO(string? workflowType) returns WorkflowInfo[]|commons:APKError {
+    postgresql:Client | error dbClient  = getConnection();
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:27]: Document is not formatted correctly

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                 FROM WORKFLOWS WHERE wf_status = 'CREATED' AND wf_type = ${workflowType};`;
+            stream<WorkflowInfo, sql:Error?> workFlowStream = dbClient->query(query);
+            check from WorkflowInfo workflow in workFlowStream do {
+                workflowList.push(workflow);
+            };
+            return workflowList;
+        } on fail var e {
+            return e909400(e);
+        }
+    }
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:44]: We can extract out this else block

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            if result is sql:ExecutionResult {
+                return payload;
+            } else {
+                return e909400(result);
+            }
+        } on fail var e {
+            return e909400(e);
+        }
+    }
+}
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:64]: Missing newline

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+isolated function getWorkflowDAO(string workflowReferenceId, WorkflowInfo payload) returns WorkflowInfo|commons:APKError {
+    postgresql:Client | error dbClient  = getConnection();
+    if dbClient is error {
+        return e909401(dbClient);
+    } else {
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:51]: Extract out the else block here as well

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        do {
+            sql:ParameterizedQuery query = `Update WORKFLOWS SET wf_status = 'COMPLETED' WHERE wf_reference = ${workflowReferenceId};`;
+            sql:ExecutionResult | sql:Error result =  dbClient->execute(query);
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:54]: not formatted correctly

File: admin/admin-domain-service/ballerina/workflowDAO.bal [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            if result is sql:ExecutionResult {
+                return payload;
+            } else {
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowDAO.bal:57]: extract out the else block

File: admin/admin-domain-service/ballerina/workflowImpl.bal [hunk 7]
```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowImpl.bal:2]:
```suggestion
// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```

File: admin/admin-domain-service/ballerina/workflowDAO.bal (hunk 6 above)
Comment (rdulmina) [reply admin/admin-domain-service/ballerina/workflowDAO.bal:57]: Please check other places as well

File: admin/admin-domain-service/ballerina/workflowImpl.bal [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        okWorkflowInfo = {
+                body: {
+                    workflowReferenceId: workflowInfo.workflowReferenceId
+                }
+            };
+        return okWorkflowInfo;
+    } else {
+        return e909400(workflowInfo);
+    }  
+}
```
Comment (rdulmina) [inline admin/admin-domain-service/ballerina/workflowImpl.bal:48]: missing new line

File: admin/admin-domain-service/ballerina/admin-api_service.bal [hunk 9]
```diff
@@ -538,8 +539,9 @@ service /api/am/admin on ep0 {
     # OkWorkflowInfo (OK. Workflow request information is returned.)
     # BadRequestError (Bad Request. Invalid request or validation error.)
     # NotFoundError (Not Found. The specified resource does not exist.)
-    // resource function post workflows/'update\-workflow\-status(string workflowReferenceId, @http:Payload WorkflowInfo payload) returns OkWorkflowInfo|BadRequestError|NotFoundError {
-    // }
+    resource function post workflows/'update\-workflow\-status(string workflowReferenceId, @http:Payload WorkflowInfo payload) returns OkWorkflowInfo|commons:APKError {
```
[the remark this answers is not in the document]
Comment (Krishanx92) [reply admin/admin-domain-service/ballerina/admin-api_service.bal]: This is ballerina code gen function

**Pull request merged**